internal/cli/ops_manager_clusters_indexes_create.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"fmt"
+	"strings"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	om "github.com/mongodb/go-client-mongodb-ops-manager/opsmngr"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/description"
+	"github.com/mongodb/mongocli/internal/flags"
+	"github.com/mongodb/mongocli/internal/messages"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type opsManagerClustersIndexesCreateOpts struct {
+	globalOpts
+	name            string
+	db              string
+	collection      string
+	rsName          string
+	locale          string
+	caseFirst       string
+	alternate       string
+	maxVariable     string
+	strength        int
+	caseLevel       bool
+	numericOrdering bool
+	normalization   bool
+	backwards       bool
+	unique          bool
+	sparse          bool
+	background      bool
+	keys            []string
+	store           store.AutomationPatcher
+}
+
+func (opts *opsManagerClustersIndexesCreateOpts) init() error {
+	if opts.ProjectID() == "" {
+		return errMissingProjectID
+	}
+
+	var err error
+	opts.store, err = store.New()
+	return err
+}
+
+func (opts *opsManagerClustersIndexesCreateOpts) Run() error {
+	current, err := opts.store.GetAutomationConfig(opts.ProjectID())
+	if err != nil {
+		return err
+	}
+
+	index, err := opts.newIndex()
+	if err != nil {
+		return err
+	}
+
+	current.IndexConfigs = index
```

Comment:
this is always empty right? we may need to consider that another operation is in progress and we may need to add and not do a replace

Reply:
mh.. actually here I am not sure. I think that is not empty when we have already defined an index and we try to add a new one. 
In my ops manager I have this behavior: after creating a new index I see it in the `indexConfig` but I do not see the others that I have already created even if they are still under the index section. When I reload the automation config after a few minutes the `indexConfig`  is empty again but all my indexes are still inside the index section.

Reply:
is this behavior normal? It does not make sense to me...

Reply:
It's kind of normal, the way it works is that it only keeps the new index to create, OM will not try to keep this list consistent, as in, if you create it here, and then go and delete it, OM will not try to recreate it (hence why it goes away)

I'm more concern on multiple index creations happening in parallel and being overwritten but it may be an edge case this is why for now this may work but we should try to make this a bit on the safer side later